### PR TITLE
feat(contracts): separate validator kro from delegation pool

### PR DESCRIPTION
**Pull request by 0xHansLee**

### CODE REVIEW

File: packages/contracts/contracts/L1/interfaces/IAssetManager.sol [hunk 1]
```diff
@@ -9,12 +9,12 @@ interface IAssetManager {
     /**
      * @notice Represents the asset information of the vault of a validator.
      *
-     * @custom:field totalKro           Total amount of KRO in the vault.
+     * @custom:field totalKro           Total amount of KRO of delegators in the vault.
```
Comment (seolaoh) [inline packages/contracts/contracts/L1/interfaces/IAssetManager.sol:12]:
```suggestion
     * @custom:field totalKro           Total amount of KRO of delegators in the vault.
     *                                  (including totalKroInKgh)
```

File: packages/contracts/contracts/L1/interfaces/IAssetManager.sol [hunk 2]
```diff
@@ -108,6 +108,14 @@ interface IAssetManager {
         mapping(address => KghDelegator) kghDelegators;
     }
 
+    /**
+     * @notice Emitted when validator deposit KROs
+     *
+     * @param validator Address of the validator.
+     * @param amount    The amount of KRO deposited.
+     */
+    event Deposited(address indexed validator, uint128 amount);
```
Comment (seolaoh) [inline packages/contracts/contracts/L1/interfaces/IAssetManager.sol:117]: How about this?
```suggestion
    event KroDeposited(address indexed validator, uint128 amount);
```

File: packages/contracts/contracts/L1/interfaces/IAssetManager.sol [hunk 3]
```diff
@@ -108,6 +108,14 @@ interface IAssetManager {
         mapping(address => KghDelegator) kghDelegators;
     }
 
+    /**
+     * @notice Emitted when validator deposit KROs
```
Comment (seolaoh) [inline packages/contracts/contracts/L1/interfaces/IAssetManager.sol:112]:
```suggestion
     * @notice Emitted when validator deposited KROs.
```

File: packages/contracts/contracts/L1/interfaces/IAssetManager.sol [hunk 4]
```diff
@@ -9,12 +9,12 @@ interface IAssetManager {
...
      * @notice Represents the asset information of the vault of a validator.
      *
-     * @custom:field totalKro           Total amount of KRO in the vault.
+     * @custom:field totalKro           Total amount of KRO of delegators in the vault.
      * @custom:field totalKroShares     Total shares for KRO delegation in the vault.
      * @custom:field totalKgh           Total number of KGH in the vault.
      * @custom:field totalKroInKgh      Total amount of KRO which KGHs in the vault have.
      * @custom:field totalKghShares     Total shares for KGH delegation in the vault.
-     * @custom:field validatorKro       Amount of KRO that the validator self-delegated.
+     * @custom:field validatorKro       Amount of KRO that the validator deposited.
```
Comment (seolaoh) [inline packages/contracts/contracts/L1/interfaces/IAssetManager.sol:17]: Could you remove `self-delegate` also in the comments of `totalValidatorKro`, `ValidatorRegistered`, and `registerValidator`?

File: packages/contracts/contracts/test/AssetManager.t.sol [hunk 5]
```diff
@@ -292,117 +293,117 @@ contract AssetManagerTest is ValidatorSystemUpgrade_Initializer {
...
-            validator,
-            delegator3
-        );
-
-        // Total KRO was 20,000 and 20 KRO was undelegated. So the reward that delegator3 can take
-        // is 20 * 20 / 20,000 = 0.02.
-        assertEq(assetManager.totalKroAssets(validator), 19999980000000000000001);
-        assertEq(pendingAssets, 20019999999999999999);
-    }
+    //    function test_initUndelegate_succeeds() public {
```
Comment (seolaoh) [inline packages/contracts/contracts/test/AssetManager.t.sol:296]: Could you just uncomment the failed tests? Let's just ignore it and fix at once.

File: packages/contracts/contracts/L1/AssetManager.sol [hunk 6]
```diff
@@ -105,10 +105,8 @@ contract AssetManager is ISemver, IERC721Receiver, IAssetManager {
      */
     modifier isRegistered(address validator) {
         if (
-            msg.sender != validator &&
-            (VALIDATOR_MANAGER.getStatus(validator) <
-                IValidatorManager.ValidatorStatus.REGISTERED ||
-                VALIDATOR_MANAGER.inJail(validator))
+            VALIDATOR_MANAGER.getStatus(validator) < IValidatorManager.ValidatorStatus.REGISTERED ||
```
[the remark this answers is not in the document]
Comment (0xHansLee) [reply packages/contracts/contracts/L1/AssetManager.sol:108]: Ah I see. With this change, the validator can self-delegate. I will rollback as you mentioned

[the remark this answers is not in the document]
Comment (0xHansLee) [reply packages/contracts/contracts/L1/AssetManager.sol:108]: But I think we need to check if the validator is registered or not in `deposit` function even if `isRegistered` modifier is not used.

File: packages/contracts/contracts/L1/AssetManager.sol [hunk 7]
```diff
@@ -105,8 +105,8 @@ contract AssetManager is ISemver, IERC721Receiver, IAssetManager {
      */
     modifier isRegistered(address validator) {
         if (
-            VALIDATOR_MANAGER.getStatus(validator) < IValidatorManager.ValidatorStatus.REGISTERED ||
-            VALIDATOR_MANAGER.inJail(validator)
+            (VALIDATOR_MANAGER.getStatus(validator) <
+                IValidatorManager.ValidatorStatus.REGISTERED) || VALIDATOR_MANAGER.inJail(validator)
```
Comment (seolaoh) [inline packages/contracts/contracts/L1/AssetManager.sol]:
```suggestion
            VALIDATOR_MANAGER.getStatus(validator) < IValidatorManager.ValidatorStatus.REGISTERED ||
            VALIDATOR_MANAGER.inJail(validator)
```

File: packages/contracts/contracts/L1/AssetManager.sol [hunk 8]
```diff
@@ -903,6 +905,14 @@ contract AssetManager is ISemver, IERC721Receiver, IAssetManager {
...
     }
 
+    /**
+     * @notice Internal function to deposit KRO to the validator.
+     *
+     * @param validator  Address of the validator.
+     * @param assets     The amount of KRO to delegate.
+     * @param updateTree Flag to update the validator tree.
+     */
+
```
Comment (seolaoh) [inline packages/contracts/contracts/L1/AssetManager.sol:915]:
```suggestion
```

File: packages/contracts/contracts/L1/AssetManager.sol [hunk 9]
```diff
@@ -903,6 +905,14 @@ contract AssetManager is ISemver, IERC721Receiver, IAssetManager {
             );
     }
 
+    /**
+     * @notice Internal function to deposit KRO to the validator.
+     *
+     * @param validator  Address of the validator.
+     * @param assets     The amount of KRO to delegate.
```
Comment (seolaoh) [inline packages/contracts/contracts/L1/AssetManager.sol:912]:
```suggestion
     * @param assets     The amount of KRO to deposit.
```

File: packages/contracts/contracts/L1/AssetManager.sol [hunk 10]
```diff
@@ -903,6 +905,14 @@ contract AssetManager is ISemver, IERC721Receiver, IAssetManager {
             );
     }
 
+    /**
+     * @notice Internal function to deposit KRO to the validator.
```
Comment (seolaoh) [inline packages/contracts/contracts/L1/AssetManager.sol:909]:
```suggestion
     * @notice Internal function to deposit KRO by the validator.
```

**Pull request merged**